src/panel/direction/DirectionInput.jsx

```diff
@@ -0,0 +1,109 @@
+/* global _ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import NavigatorGeolocalisationPoi, { navigatorGeolocationStatus } from
+  'src/adapters/poi/specials/navigator_geolocalisation_poi';
+import Suggest from 'src/adapters/suggest';
+
+export default class DirectionInput extends React.Component {
+  static propTypes = {
+    value: PropTypes.string,
+    onChangePoint: PropTypes.func.isRequired,
+    pointType: PropTypes.oneOf(['origin', 'destination']).isRequired,
+    claimFocus: PropTypes.bool,
+  }
+
+  constructor(props) {
+    super(props);
+    this.inputRef = React.createRef();
+  }
+
+  componentDidMount() {
+    this.suggest = new Suggest({
+      tagSelector: `#itinerary_input_${this.props.pointType}`,
+      onSelect: this.selectItem,
+      prefixes: [ NavigatorGeolocalisationPoi.getInstance() ],
+      menuClass: 'direction_suggestions',
+    });
+    if (this.props.claimFocus) {
+      this.focus();
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.props.claimFocus) {
+      this.focus();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.suggest) {
+      this.suggest.destroy();
+    }
+  }
+
+  onChange = event => {
+    const input = event.target.value;
+    this.props.onChangePoint(input, null);
+    this.suggest.setValue(input);
```

Comment:
As far as I understand, this function is called on each keystroke.
And as a consequence, the suggestions list is flickering between requests.
Maybe due to how `suggest.setValue` resets the suggestions list ?

Actually this looks redundant with the existing mechanism in Autocomplete that creates listeners on "input" events.

Reply:
Simply removing the line solved the problem, meaning the change handler registered by the autocomplete widget continues to work fine on a React-controlled input.